xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java

```diff
@@ -302,4 +293,25 @@ private String computeAltAttributeValue(ResourceReference reference)
         }
         return label;
     }
+
+    private boolean shouldIgnoreAll(TagStack stack)
+    {
+        Optional<Iterator<Object>> stackParameterIterator =
+            Optional.ofNullable(stack.getStackParameterIterator(NON_GENERATED_CONTENT_STACK));
+        boolean inNonGeneratedContent =
+            stackParameterIterator
+                .map(iterator -> {
+                    while (iterator.hasNext()) {
+                        NonGeneratedContentStackValue value = (NonGeneratedContentStackValue) iterator.next();
+                        // Stops at the first value which is associated with a macro.
+                        if (value.isMetaDataElement) {
+                            return value.nonGeneratedContent;
+                        }
+                    }
+                    return false;
+                }).orElse(false);
+
+        // If we are in a macro but not in a non generated content, we should ignore all
+        return stack.getStackParameter(MACRO_INFO) != null && !inNonGeneratedContent;
```

Comment:
Doesn't every div push a new stack and thus "remove" the `MACRO_INFO` stack parameter?

Reply:
Does it?

Reply:
See https://github.com/xwiki/xwiki-rendering/blob/045ae68924e6bc80fb2ccea17f64349817244c06/xwiki-rendering-wikimodel/src/main/java/org/xwiki/rendering/wikimodel/impl/WikiScannerContext.java#L149 and https://github.com/xwiki/xwiki-rendering/blob/045ae68924e6bc80fb2ccea17f64349817244c06/xwiki-rendering-wikimodel/src/main/java/org/xwiki/rendering/wikimodel/impl/WikiScannerContext.java#L704

Reply:
Thanks. So I need to find a way to visit the stack of macro calls, and not the internal state of a single macro.